### PR TITLE
Common base for Blockbook & Blockfrost websocket

### LINKED ISSUE #8946: Blockchain-link: unify websocket logic
Now there are some duplicities in websocket implementation between blockfrost and blockbook. Lets get rid of the 🙏

### PR DESCRIPTION
**Pull request by marekrjpolak**
## Description

Removed duplicate code from blockbook and blockfrost websocket implementation.

## Related Issue

Resolve #8946

### CODE REVIEW

File: packages/blockchain-link/src/workers/apiBase.ts [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+const DEFAULT_PING_TIMEOUT = 50 * 1000;
+
+type EventMap = { [event: string]: any };
+
+type WsEvents = {
+    error: string;
+    disconnected: undefined;
+};
+
+export abstract class APIBase<T extends EventMap> extends TypedEmitter<T & WsEvents> {
```
Comment (szymonlesisz) [inline packages/blockchain-link/src/workers/apiBase.ts]: there is already `base.ts` (BaseWorker) file in workers dir. maybe lets name it `BaseWebsocket` ?

Comment (mroz22) [reply packages/blockchain-link/src/workers/apiBase.ts]: maybe also renaming those files `baseWorker.ts -> BaseWorker class` and `baseWebsocket.ts -> BaseWebsocket class` could help when navigating through the files.

**Review by mroz22 — APPROVED** [review]
works on my machine and removes code :shipit:

File: packages/blockchain-link/src/workers/apiBase.ts (hunk 1 above)
Comment (marekrjpolak) [reply packages/blockchain-link/src/workers/apiBase.ts]: Both of you are right, fixed & squashed.

**Pull request merged**